### ISSUE TITLE
[Explore]: Package upload - Source code to Image conversion

### ISSUE THREAD

**Issue by akrishna90** [cloudfoundry/cf-k8s-api#36]
### Background

**Blocked By** 
[[Explore]: Packaging source code as Carvel's imgpkg ](https://github.com/cloudfoundry/cf-k8s-api/issues/35)

Based on the recommendation from #35, explore on how to convert source code into the recommended image format.

**Things to remember**

- Ease of Scalability
- Performance

### Acceptance Criteria

Document the design, decisions and reasoning in an ADR

**Comment by Birdrock** [comment]
Superseded by #46
